### PR TITLE
[FreeBSD 14] The default nvme disk name is changed in FreeBSD 14 which cause the failure for autoinstall

### PR DESCRIPTION
**Pull request by ZouYuhua**
The nvme disk name in FreeBSD 13 is nvd0, but in FreeBSD it is changed to nda0.
We need to change autoinstall for FreeBSD 14.

Test result:
image

### CODE REVIEW

**Comment by ZouYuhua** [comment]
@keirazhang  Please help codereview it.  Thanks

File: autoinstall/FreeBSD/installerconfig [hunk 1]
```diff
@@ -7,7 +7,14 @@ export ZFSBOOT_SWAP_SIZE=2g
 export nonInteractive="YES"
 export ZFSBOOT_CONFIRM_LAYOUT=0
 
-{% if boot_disk_controller == 'nvme' %}
+{% set freebsd_major_version = 14 %}
+{% if guest_os_type is defined and guest_os_type %}
```
Comment (keirazhang) [inline autoinstall/FreeBSD/installerconfig:11]: where did you define `guest_os_type`? If there is no good way to identify FreeBSD version before deploying vm. I think you can add a new installerconfig for FreeBSD 14. Move the original one as `autoinstall/FreeBSD/13/installerconfig`, and add a new one for FreeBSD 14: `autoinstall/FreeBSD/14/installerconfig`. The only difference between the two installerconfig is the value of `ZFSBOOT_DISKS`.

**Comment by ZouYuhua** [comment]
@keirazhang 
Passed after code change:
image

Thanks

**Review by keirazhang — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**